src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -187,10 +190,120 @@ public void delete(Long examId) {
         return exams.stream().filter(exam -> Boolean.TRUE.equals(exam.isVisibleToStudents())).collect(Collectors.toSet());
     }
 
+    /**
+     * Returns the relevant result of a student participation
+     *
+     * @param studentParticipation studentParticipation to get relevant result for
+     * @return optional of relevant result
+     */
+    private Optional<Result> getRelevantResult(StudentParticipation studentParticipation) {
+        // no participant -> no relevant result
+        if (studentParticipation.getParticipant() == null) {
+            return Optional.empty();
+        }
+
+        return studentParticipation.getResults().stream().filter(Result::isRated).filter(result -> result.getCompletionDate() != null).filter(result -> result.getScore() != null)
+                .sorted((r1, r2) -> r2.getCompletionDate().compareTo(r1.getCompletionDate())).findFirst();
+
+    }
+
+    /**
+     * Puts students, result and exerciseGroups together for ExamScoresDTO
+     *
+     * @param examId the id of the exam
+     * @return return ExamScoresDTO with students, scores and exerciseGroups for exam
+     */
+    public ExamScoresDTO getExamScore(Long examId) {
+        Exam exam = examRepository.findForScoreCalculationById(examId).orElseThrow(() -> new EntityNotFoundException("Exam with id: \"" + examId + "\" does not exist"));
+
+        // Adding exam information to DTO
+        ExamScoresDTO scores = new ExamScoresDTO(exam.getId(), exam.getTitle(), exam.getMaxPoints());
+
+        // Adding exercise group information to DTO
+        for (ExerciseGroup exerciseGroup : exam.getExerciseGroups()) {
+            // Alert: This only works if all exercises in an exercise groups have the same number of maximum points
+            Double maximumNumberOfPoints = null;
+            if (!exerciseGroup.getExercises().isEmpty()) {
+                maximumNumberOfPoints = exerciseGroup.getExercises().iterator().next().getMaxScore();
+            }
+
+            List<String> containedExercises = new ArrayList<>();
+
+            for (Exercise exercise : exerciseGroup.getExercises()) {
+                containedExercises.add(exercise.getTitle().trim());
```

Comment:
Why do we use an title here instead of an ID?

Reply:
I think Stefan intended to send only necessary data. This is one reason why he used DTOs here.

Reply:
I guess these are used for the columns in the export, but I can't tell you for sure

Reply:
I just looked it up, this is used to display the title of the exercise in the html.
There is no other usage, so I think it should be fine!